x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -87,8 +89,11 @@ export const useUnifiedSearch = () => {
     if (!metricsDataView) {
       return null;
     }
-    return buildEsQuery(metricsDataView, state.query, state.filters);
-  }, [metricsDataView, state.filters, state.query]);
+    if (panelFilters && panelFilters.length > 0) {
```

Comment:
**nit**: `if (panelFilters?.length > 0)`

Reply:
In this case, `panelFilters` can be `null` - So if it's null the filters are not in the URL, it can be an empty array if there are no filters and an array with filter objects: so the type is `Filter[] | null` and type check is not passing if I change it to `panelFilters?.length`. 
So to make it clear I used `Array.isArray(panelFilters) && panelFilters.length > 0`.